benchmark/metrics_parser.py

```diff
@@ -0,0 +1,276 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+import argparse
+from collections import defaultdict
+import csv
+import itertools
+import json
+
+
+WALL_CLOCK_KEY = u'-- WALL --'
+
+
+HEADERS = [
+    u'Flocker Version',
+    u'Nodes',
+    u'Containers',
+    u'Scenario',
+    u'Control Service CPU',
+    u'Dataset Agent CPU',
+    u'Container Agent CPU',
+    u'Containers Converged Within Limit',
+    u'Scenario Requests Within Limit',
+]
+
+
+def write_csv(results, filename):
+    with open(filename, 'w') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=HEADERS)
+        writer.writeheader()
+        writer.writerows(results)
+
+
+def mean(values):
+    if len(values) > 0:
+        return sum(values) / len(values)
+    return None
+
+
+def cputime_for_process(results, process):
+    """
+    Calculate the CPU time for a process running in a particular scenario.
+
+    By default this will return the `mean` result.
+    """
+    process_results = itertools.ifilter(
+        lambda r: r['metric']['type'] == 'cputime' and r['process'] == process,
+        results
+    )
+    values = [r['value'] / r['wallclock'] for r in process_results]
+    return mean(values)
+
+
+def wallclock_for_operation(results, operation):
+    """
+    Calculate the wallclock time for a process running in a particular
+    scenario.
+
+    By default this will return the `mean` result.
+    """
+    operation_results = itertools.ifilter(
+        lambda r: r['metric']['type'] == 'wallclock'
+        and r['operation']['type'] == operation,
```

Comment:
Put this and other broken `and` operators at end of previous line - see https://www.python.org/dev/peps/pep-0008/#id18